src/main/drivers/transponder_ir_arcitimer.c

```diff
@@ -36,7 +36,7 @@ void transponderIrInitArcitimer(transponder_t *transponder){
     transponder->vTable             = &arcitimerTansponderVTable;
     transponder->timer_hz           = TRANSPONDER_TIMER_MHZ_ARCITIMER;
     transponder->timer_carrier_hz   = TRANSPONDER_CARRIER_HZ_ARCITIMER;
-    memset(&(transponder->transponderIrDMABuffer.arcitimer), 0, TRANSPONDER_DMA_BUFFER_SIZE_ARCITIMER);
+    memset(&(transponder->transponderIrDMABuffer.arcitimer), 0, sizeof(transponder->transponderIrDMABuffer.arcitimer));
```

Comment:
I think this is unrelated to gcc. Should this go into a separate pull request? Also, is this define now unused and should it be removed?

Reply:
why this change is needed are descriped in the pull request message.
also of coure the marko is still in use. transponderIrDMABuffer uses this to determine the array size

Reply:
Still, it's a change that is unrelated to changing the gcc version, so it should be a separate pull request.

Reply:
I dont see how this is a unrelated change. the gcc change result in a warning of this line.

So we should merge the change to the gcc version. having compiler warnings and then i could create another pr which fixes those warnings?
I think a change that result in a warning and the removing of this warning should be in the same pull request, so we see why those changes happen.

maybe we could get another opinion here since we both disagree?
@AndersHoglund @DieHertz?

Reply:
Sorry for a late and outdated reply, changes have already been split to several PRs. 
I am of the opinion that you should only do one change per PR, it should be complete and atomic taking care of all its dependencies and side effects. Like in this case, a new compiler that uncovers some old hidden issues and also has new requirements. All of this could be done in one and the same PR, as it was originally done. Preparing a new baseline by first doing a number of separate PRs and then do the actual update is only possible by first doing it the other way to see the problems.  Only then PRs for base line updates can be made, but those can only be tested by doing the actual gcc update... Do you see the possible iteration here? Unnecessary iteration with multiple PRs involved. One complete atomic PR would be much cleaner. But it is all history now, its over and done. 

But sure, two or more completely unrelated things should never ever be done in one PR. I have seen too many such things. First a functional change, and also "by the way" a whole lot of unrelated cosmetic touch-ups. That only obscures the real change when you run a diff.